library/scenemanager.js

```diff
@@ -0,0 +1,198 @@
+//
+// p5 SceneManager helps you create p5.js sketches with multiple states / scenes
+// Each scene is a like a sketch within the main sketch. You focus on creating
+// the scene like a regular sketch and SceneManager ensure scene switching
+// routing the main setup(), draw(), mousePressed(), etc. events to the
+// appropriate current scene.
+//
+// Author: Marian Veteanu
+// http://github.com/mveteanu
+//
+function SceneManager(p) {
+  this.scenes = [];
+  this.scene = null;
+
+  // Wire relevant p5.js events, except setup()
+  // If you don't call this method, you need to manually wire events
+  this.wire = function() {
+    const P5Events = [
+      "mouseClicked",
+      "mousePressed",
+      "mouseReleased",
+      "mouseMoved",
+      "mouseDragged",
+      "doubleClicked",
+      "mouseWheel",
+      "keyPressed",
+      "keyReleased",
+      "keyTyped",
+      "touchStarted",
+      "touchMoved",
+      "touchEnded",
+      "deviceMoved",
+      "deviceTurned",
+      "deviceShaken"
+    ];
+
+    var me = this;
+    var o = p != null
```

Comment:
this needs to be much clearer. what is o what is p
use meaningful variable names

Reply:
This is a library, it isn't code I wrote